module/VuFind/src/VuFind/Db/Table/Resource.php

```diff
@@ -55,41 +54,33 @@ class Resource extends Gateway implements DbServiceAwareInterface
     use DbServiceAwareTrait;
 
     /**
-     * Date converter
+     * Loader for record populator
      *
-     * @var DateConverter
+     * @var callable
      */
-    protected $dateConverter;
-
-    /**
-     * Record loader
-     *
-     * @var Loader
-     */
-    protected $recordLoader;
+    protected $resourcePopulatorLoader;
 
     /**
      * Constructor
      *
-     * @param Adapter       $adapter   Database adapter
-     * @param PluginManager $tm        Table manager
-     * @param array         $cfg       Laminas configuration
-     * @param RowGateway    $rowObj    Row prototype object (null for default)
-     * @param DateConverter $converter Date converter
-     * @param Loader        $loader    Record loader
-     * @param string        $table     Name of database table to interface with
+     * @param Adapter       $adapter                 Database adapter
+     * @param PluginManager $tm                      Table manager
+     * @param array         $cfg                     Laminas configuration
+     * @param RowGateway    $rowObj                  Row prototype object (null for default)
+     * @param DateConverter $dateConverter           Date converter
+     * @param callable      $resourcePopulatorLoader Resource populator loader
+     * @param string        $table                   Name of database table to interface with
      */
     public function __construct(
         Adapter $adapter,
         PluginManager $tm,
         $cfg,
         ?RowGateway $rowObj,
-        DateConverter $converter,
-        Loader $loader,
+        protected DateConverter $dateConverter,
+        callable $resourcePopulatorLoader,
         $table = 'resource'
```

Comment:
Could we add the missing types (excluding `callable`) as the constructor is being changed?